lib/pbench/test/unit/common/test_config.py

```diff
@@ -1,68 +1,173 @@
+        PbenchConfig uses those values instead of default values"""
+
+        logger_type = "file"
+        log_dir = "/mock/log_dir"
+        logging_level = "Devlog"
+        log_format = "1970-01-01T00:00:42.000000 -- 'message'"
+        self.logger_test_setup(
+            monkeypatch,
+            {
+                "logging": {
+                    "logger_type": logger_type,
+                    "log_dir": log_dir,
+                    "logging_level": logging_level,
+                    "log_format": log_format,
+                }
+            },
+        )
+        config = PbenchConfig("pbench.cfg")
+        assert config.logger_type == logger_type
+        assert config.log_dir == log_dir
+        assert config.log_using_caller_directory is False
+        assert config.default_logging_level == logging_level
+        assert config.log_fmt == log_format
+        assert config.TZ == "UTC"
+
+    @pytest.mark.parametrize(
+        "config_value, expected_error_msg",
+        [
+            (
+                {"logging": {"logger_type": "hostport"}},
+                "No option 'logger_host' in section: 'logging'",
+            ),
+            (
+                {
+                    "logging": {
+                        "logger_type": "hostport",
+                        "logger_host": "logger.example.com",
+                    }
+                },
+                "No option 'logger_port' in section: 'logging'",
+            ),
+        ],
+    )
+    def test_logger_hostport_BadConfig(
+        self, config_value, expected_error_msg, monkeypatch
+    ):
+        """Show that PbenchConfig raises 'BadConfig' when the
+        configuration does not contain required option."""
+
+        self.logger_test_setup(monkeypatch, config_value)
+        with pytest.raises(BadConfig) as exc:
+            PbenchConfig("pbench.cfg")
+        assert expected_error_msg in str(exc.value)
+
+    @pytest.mark.parametrize(
+        "config_value, expected_error_msg",
+        [
+            (
+                {"logging": {"logger_type": "hostport", "logger_host": ""}},
+                "ValueError('logger_host', 'logging')",
+            ),
+            (
+                {
+                    "logging": {
+                        "logger_type": "hostport",
+                        "logger_host": "logger.example.com",
+                        "logger_port": "",
+                    }
+                },
+                "ValueError('logger_port', 'logging')",
+            ),
+        ],
+    )
+    def test_logger_hostport_ValueError(
+        self, config_value, expected_error_msg, monkeypatch
+    ):
+        """Show that PbenchConfig raises 'ValueError' when the
+        option contains an empty value."""
+
+        self.logger_test_setup(monkeypatch, config_value)
+        with pytest.raises(ValueError) as exc:
+            PbenchConfig("pbench.cfg")
+        assert expected_error_msg in str(exc)
```

Comment:
Since the only difference (that I can spot) between `test_logger_hostport_BadConfig()` and `test_logger_hostport_ValueError()` is the exception passed to the `with` statement, I would add that to the parametrize-ation and combine the two test cases together.